### ISSUE TITLE
HMR and lazyView

### ISSUE THREAD

**Issue by toburger** [elmish/hmr#8]
### Description

Just wanted to continue the discussion from https://github.com/elmish/react/issues/22 and the somehow related issue from https://github.com/elmish/hmr/issues/6.

### Repro code

Usage of both packages `Fable.Elmish.HMR` and `Fable.Elmish.React` combined result in subparts of the view not updated when using HMR and one of the `lazyX` functions from Elmish.React.

#### Expected:

One of two behaviors:

* Both modules work in combination without modification of users code.
* When using both modules in combination give a meaningful warning/error message that the combination of both modules isn't supported OOB.

#### Actual:
Certain parts of the view aren't updated and it is hard for the consumer of the two libs to figure out why views aren't updated anymore.

**Comment by MangelMaxime** [comment]
Do you have additional ideas or comments on the ideas we exposed at the end of the discussion?

**Comment by toburger** [comment]
No better idea other than your proposal of shadowing the lazyX functions, sorry...

**Comment by MangelMaxime** [comment]
@et1975 

What do you think of this proposition (copied from https://github.com/elmish/react/issues/22#issuecomment-374988783):

I believe the easiest way is to use the `HMR.Common` shadowing.

Module code would be something like:

```fs
module HMR.Common

let inline lazyView x = x

let inline lazyViewWith cmp x = x

// Etc.
```

User code:

```fs
#if HMR
open HMR.Common
#endif
```

And also adding the `HMR`constant to webpack config.

**Comment by MangelMaxime** [comment]
I think I know how to fix `lazyView` usage with HMR support. But then it's possible that react stateful can have some special behavior on some circumstance. I will need to explore my idea and make some tests before being able to share the code in the open.

**Comment by et1975** [comment]
The shadowing should happen by default IMHO, there shouldn't be a need to discover correct solution. Just opening the `Elmish.HMR` should bring in the shadows for `lazyView`,`Program.toNavigable`, etc.
The order of module openings is a concern, but if you get it right in the template - you've covered 80% of cases, and the last 20% can find the solution in the docs.

**Comment by MangelMaxime** [comment]
Yes I think because Fable is compiling the source code we can use `#if DEBUG` in the lib code directly and not ask the user to do it for us.

Like that he just need to do:

```fs
open Elmish.Common
open Elmish.HMR

let view = 
   lazyView ....
```

And so here `lazyView` will comes from `Elmish.HMR.lazyView` in `Debug` mode and from `Elmish.Common.lazyView` in production code.

**Comment by MangelMaxime** [comment]
This has been fixed by #11 

Use the following `open` statement and you will have it working.

```fs
open Elmish.React
open Elmish.HMR
```

The `lazyView` content is forced to refresh on HMR trigger but otherwise still behave as a `lazyView`.

Demo:

![2018-11-09 21 40 44](https://user-images.githubusercontent.com/4760796/48423242-88097d00-e760-11e8-8575-7c9221e15292.gif)
